java/arcs/sdk/wasm/WasmSingleton.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.sdk.wasm
+
+import arcs.sdk.ReadWriteSingleton
+
+open class WasmSingleton<T : WasmEntity>(
```

Comment:
KDoc? Also, for this too - consider locking on a mutex (or mutltiple) as necessary for thread-safety?

Reply:
KDoc added. Will punt on the mutex stuff